### ISSUE TITLE
How to dynamically log HTTP requests/responses on the wire?

### ISSUE THREAD

**Issue by kodeninja** [denoland/deno#9985]
Is there a way to dynamically log all HTTP requests/responses (headers etc) for `fetch` calls? I tried running with `RUST_LOG="debug"` as well as `RUST_LOG="trace"` but that didn't help.

I couldn't find any documentation on this.

**Comment by lucacasonato** [comment]
No, this is not possible at the moment. You can inspect the count of `fetch` calls in `--unstable` using `Deno.metrics()`

**Comment by kodeninja** [comment]
I see. Thanks, @lucacasonato.

**Comment by danopia** [comment]
One workaround is to reassign `globalThis.fetch` to a custom function in your program's entrypoint. I've done this for metrics purposes and nothing's broke so far

**Comment by kodeninja** [comment]
@danopia, interesting! I'll give it a try 🙂

**Comment by bartlomieju** [comment]
@lucacasonato probably not something we want to add given WPT tests?

**Comment by danopia** [comment]
Does WPT mandate that fetch requests go directly to the server? Web browsers let fetch() get hooked by registering the fetch event from a ServiceWorker, for example

**Comment by lucacasonato** [comment]
@danopia Yeah, but service workers are incredibly complicated, and I don't see us implementing them any time soon.

**Comment by danopia** [comment]
Sure; I don't see how WPT would block a simpler mechanism as was implied, though.
